tests/e2e-cypress/integration/drawing/measureOverlays.cy.js

```diff
@@ -35,4 +87,70 @@ describe('Measure Overlays handling', () => {
+     * - That the measureManager of the drawingLayer correctly removes all overlays and that the
+     *   measureManager of the kml layer correctly takes over
+     */
+    it('Check correct passover from drawingLayer to kmlLayer when closing drawing', () => {
+        //Open drawing mode
+        cy.goToDrawing(language, { lat: latitude, lon: longitude, z: zoom }, true)
+        cy.readWindowValue('map')
+            .then((map) => map.getOverlays().getLength())
+            .as('nbOverlaysBeforeDrawing')
+        cy.readWindowValue('drawingLayer')
+            .then((layer) => layer.getSource().getFeatures())
+            .should('have.length', 0)
+        addKmlInterceptAndReinject()
+        //Draw a measure
+        cy.clickDrawingTool(EditableFeatureTypes.MEASURE)
+        cy.get(olSelector).click('left')
+        cy.get(olSelector).click('center')
+        cy.get(olSelector).dblclick('center')
+
+        //Overlays should be visible
+        cy.readStoreValue('state.layers.activeLayers').should('have.length', 0)
+        cy.readWindowValue('map')
+            .then((map) => map.getOverlays().getLength())
+            .as('nbOverlaysAfterDrawing')
+            .should(function (nb) {
+                expect(nb).to.be.greaterThan(this.nbOverlaysBeforeDrawing)
+            })
+            .then(function (nb) {
+                cy.wrap(nb - this.nbOverlaysBeforeDrawing).as('nbOverlaysDrawn')
+            })
+
+        //Close drawing mode and check that the same number of the features and Overlays are
+        //displayed
+        cy.log('Close drawing mode')
+        cy.get('[data-cy="drawing-toolbox-close-button"]').click()
+        cy.readStoreValue('state.layers.activeLayers').should('have.length', 1)
+        cy.readWindowValue('map').should(function (map) {
+            const length = map.getOverlays().getLength()
+            expect(length).to.be.equal(this.nbOverlaysDrawn + this.nbOverlaysAtBeginning)
+        })
+        cy.readWindowValue('kmlLayer').should((layer) => {
+            const features = layer.getSource().getFeatures()
+            expect(features.length).to.be.equal(1)
+        })
+
+        //Hide KML layer and check that kml layer and overlays disappeared
+        cy.fixture('service-kml/create-file.fixture.json').as('fileFixture')
+        cy.readWindowValue('kmlLayer').then(function (layer) {
+            const kmlUrl = layer.getSource().getUrl()
+            const kmlLayerSelector =
+                `[data-cy="button-toggle-visibility-` +
+                `layer-KML|${kmlUrl}|${i18n.global.t('draw_layer_label')}` +
+                `@adminId=${this.fileFixture.admin_id}"]`
+            cy.get(kmlLayerSelector).click()
+            cy.readWindowValue('map').should(function (map) {
+                const length = map.getOverlays().getLength()
+                expect(length).to.be.eq(this.nbOverlaysAtBeginning)
+            })
+            cy.readWindowValue('kmlLayer').should('not.exist')
```

Comment:
I would prefer that we avoid using `this.` as it is the most confusing aspect of Javascript, what `this` mean in each scope.
Insead, please use this approach to get the value : https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Avoiding-the-use-of-this

Reply:
The only solution that I currently see to avoid the references to "this" would be to create nested then statements for every variable that is now defined as a cypress alias. This would create a pyramid of doom, which I think is even harder to read.
The first draft of this test was indeed without the references to "this" but with the pyramid of doom, and I decided to rewrite it to avoid this pyramid by using aliases.

Reply:
fair enough, then maybe add a little comment before you use this to clarify that this comes out of a Cypress variable defined above